cinema/views.py

```diff
@@ -118,12 +141,24 @@ class OrderPagination(PageNumberPagination):
     max_page_size = 100
 
 
-class OrderViewSet(viewsets.ModelViewSet):
+class OrderViewSet(
+    mixins.CreateModelMixin,
+    mixins.ListModelMixin,
+    viewsets.GenericViewSet,
+):
     queryset = Order.objects.prefetch_related(
         "tickets__movie_session__movie", "tickets__movie_session__cinema_hall"
     )
     serializer_class = OrderSerializer
     pagination_class = OrderPagination
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (IsAuthenticated,)
+
+    def get_permissions(self):
+        if self.action == "create":
+            return [IsAuthenticated()]
+
+        return super().get_permissions()
```

Comment:
I think, this method is redundant, because `permission_classes = (IsAuthenticated,)`